### DISCUSSION TITLE
Retries on the failure

### DISCUSSION THREAD

**Discussion by st2654** [akunzai/GSS.Authorization.OAuth#73] [General]
Hi @akunzai ,

I am planning to use the library however it's missing Retry on HTTP calls.

How do you plan to add retry?

**Comment by st2654** [comment]
PS: I am not C# pro, but working on new tech.

**Reply by akunzai** [reply]
Hi @st2654,

We don't need to modify the OAuth HTTP Client directly,

Please read [this guide](https://docs.microsoft.com/dotnet/architecture/microservices/implement-resilient-applications/implement-http-call-retries-exponential-backoff-polly)

We can easily add a retry policy with [Microsoft.Extensions.Http.Polly](https://www.nuget.org/packages/Microsoft.Extensions.Http.Polly)

For example:

```csharp
services.AddOAuth2HttpClient<ClientCredentialsAuthorizer>("oauth2",(resolver, options) =>
{
    var configuration = resolver.GetRequiredService<IConfiguration>();
    options.AccessTokenEndpoint = configuration.GetValue<Uri>("OAuth2:AccessTokenEndpoint");
    options.ClientId = configuration["OAuth2:ClientId"];
    options.ClientSecret = configuration["OAuth2:ClientSecret"];
    options.Credentials = new NetworkCredential(configuration["OAuth2:Credentials:UserName"], configuration["OAuth2:Credentials:Password"]);
    options.Scopes = configuration.GetSection("OAuth2:Scopes").Get<IEnumerable<string>>();
})
.AddPolicyHandler(HttpPolicyExtensions.HandleTransientHttpError()
.WaitAndRetryAsync(3, retryAttempt => TimeSpan.FromSeconds(30 * retryAttempt)));
```

**Reply by st2654** [reply]
Thanks, that helps. Appreciate your help and time.
